### DISCUSSION TITLE
Override field-level exclude

### DISCUSSION THREAD

**Discussion by couteau** [pydantic/pydantic#4899] [Question] [answered]
I may be missing something obvious, but I cannot find a way to include a field exporting a model (e.g., using dict()) if that field has been marked for exclusion in the model definition using the Field function. Here is a working example of what I've tried:

```
from pydantic import BaseModel, Field

class MyClass(BaseModel):
    foo: str
    bar: str | None = Field(default=None, exclude=True)

m = MyClass(foo="Foo", bar="Bar")

m.dict()                           # result: {'foo': 'Foo'} as expected
m.dict(include={"bar": True})      # result: {}, expected result: {'bar': 'Bar'}
m.dict(exclude={"bar": False})     # result: TypeError: Unexpected type of exclude value <class 'bool'>, expected result {'foo': 'Foo', 'bar': 'Bar'}
m.dict(exclude={"bar": None})      # result: {'foo': 'Foo'} 
```

In other words, once a field has been marked as excluded in the model definition, I have found no way to override that setting and include it with the model export. Is there a way to do this? Or is there some other approach (other than explicitly passing the list of fields to exclude every time I export the model, which does not work in my use case)?

**Comment by jimkring** [comment]
This is not my area of expertise, but I had an idea you might try. Can you change the configuration setting for that field and then invoke the “model_rebuild()” method on your model? I don’t recall if that’s the exact name of the method, but it’s similar. Hope that works for you and let me know.

**Reply by samuelcolvin** [reply]
`model_rebuild()` is new in V2/on main, there's no equivalent in v1.

**Comment by samuelcolvin** [comment, marked as answer]
I think this isn't possible in v1, it will be possible in V2.

**Reply by couteau** [reply]
Thank you. Looking forward to v2!

**Reply by mcndt** [reply]
How is this achieved in V2? Running into the same problem.

**Reply by kdcokenny** [reply]
same question

**Reply by ideepu** [reply]
```
In [127]: from pydantic import BaseModel, Field
     ...:
     ...: class MyClass(BaseModel):
     ...:     foo: str
     ...:     bar: str | None = Field(default=None, exclude=True)
     ...:
     ...: m = MyClass(foo="Foo", bar="Bar")

In [128]: m.model_dump()
Out[128]: {'foo': 'Foo'}

In [129]: MyClass.model_fields['bar'].exclude = False

In [130]: MyClass.model_rebuild(force=True)
Out[130]: True

In [131]: m.model_dump()
Out[131]: {'foo': 'Foo', 'bar': 'Bar'}
```


Not sure this is the right way though.

**Reply by TheCaffinatedDeveloper** [reply]
Ooof , that's rough considering if you want the fields to be excluded after the dump. Are there any plans to allow exclude arg on serialization to take priority over field level?

**Reply by VassilisVassiliadis** [reply]
Also running into this problem. 

Does this solution require us to visit all fields in the model (including nested ones) and ensure everything has `exclude=False` ? Are there any plans for what @TheCaffinatedDeveloper suggested ?

**Reply by nocturn9x** [reply]
Same issue here. Impossible to override exclusion while calling model_dump...
